segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -253,7 +279,11 @@ protected void doStart() {
     }
 
     private CompletableFuture<Void> initializeSecondaryServices() {
-        this.storage.initialize(this.metadata.getContainerEpoch());
+        try {
+            initializeStorage();
+        } catch (Exception ex) {
+            doStop(ex);
```

Comment:
I am not sure this is a good way of doing this:
- why aren't we fast failing and return a failed future?
- why are we catching a generic Exception?

Reply:
You do not need to do this. Let the exception bubble up, whatever that exception is, and observe how this method is used. In `startWhenDurableLogOnline`, a listener is attached to its returned Future which will invoke the (correct) shutdown procedure. No need for extra stuff here.

Reply:
Fixed.